src/mapping/parser/grammar.pest

```diff
@@ -0,0 +1,89 @@
+mapping = _{ SOI ~ statement ~ (NEWLINE+ ~ statement)* ~ NEWLINE* ~ EOI }
+
+statement = _{ assignment | function | if_statement }
+
+assignment = { target_path ~ "=" ~ query_arithmetic }
+
+if_statement = {
+    "if" ~ query_arithmetic ~ "{" ~ NEWLINE* ~
+        statement ~
+    NEWLINE* ~ "}" ~ ( "else" ~ "{" ~ NEWLINE* ~
+        statement ~
+    NEWLINE* ~ "}" )?
+}
+
+// Paths
+path_segment = @{ ('a'..'z' | 'A'..'Z' | "_" | "-" )+ ~ ( "[" ~ ('0'..'9')+ ~ "]" )? }
+
+target_path = @{ ("." ~ path_segment)+ }
+
+// Functions
+function = { deletion | only_fields }
+
+deletion = { "del(" ~ target_path ~ ")" }
+only_fields = { "only_fields(" ~ target_path ~ ("," ~ target_path)* ~ ")" }
+
+// Queries
+path_coalesce = !{ "(" ~ path_segment ~ ("|" ~ path_segment)+ ~ ")" }
+
+dot_path = ${ ("." ~ (path_segment | path_coalesce))+ }
+
+// Query functions
+//
+// NOTE: Not including delete here yet, but there's no reason we can't slot it
+// in at some point and get `.nested = del(.foo)`, same with `only_fields`.
+query_function = { to_string | to_int | to_float | to_bool | to_timestamp }
```

Comment:
Including specific function names in the parser probably helps with error messages, but this seems like something we'll eventually want to move to a later stage. Especially if we want to allow, for example, loadable wasm functions somewhere down the line.

Reply:
Yeah absolutely, the parser expression will be quite general so it'll need a bit of parser-fu to avoid polluting error messages but this is a logical next step.

Reply:
On the other hand, these are not normal function, as the type names could be considered reserved words.